std/machines/large_field/rotate.asm

```diff
@@ -0,0 +1,73 @@
+use std::utils::unchanged_until;
+use std::utils::cross_product;
+use std::convert::int;
+use std::check::require_field_bits;
+
+/// Rotate on single byte input using an exhaustive table, returning 32-bit value as result.
+/// TODO this way, we cannot prove anything that rotates by more than 31 bits.
+machine ByteRotate with
+    latch: latch,
+    operation_id: operation_id,
+    degree: 65536
+{
+    operation run<0> P_operation, P_A, P_B, P_ROW -> P_C;
+
+    col fixed latch = [1]*;
+    col fixed operation_id = [0]*;
+
+    require_field_bits(16, || "The field modulo should be at least 2^16 - 1 to work in the rotate machine.");
```

Comment:
Don't you need at least 32 bits if you return a 32 bit value?

Reply:
Sorry, mixed up with the small_field